multihost_test/bz_automation/conftest.py

```diff
@@ -105,3 +105,24 @@ def restore_conf():
         execute_cmd(session_multihost, "userdel -rf anuj_test")
 
     request.addfinalizer(restore_conf)
+
+
+@pytest.fixture(scope='function')
+def f_1949137(session_multihost, request):
+    """ create users for test """
+    execute_cmd(session_multihost, "useradd -u 101 systest")
+    execute_cmd(session_multihost, "cp -vf /etc/pam.d/system-auth "
+                                   "/etc/pam.d/system-auth_f_1949137")
+    file_location1 = "/multihost_test/bz_automation/script/system-auth"
+    # file_location1 = "/system-auth"
+    session_multihost.client[0].transport.put_file(os.getcwd() +
+                                                   file_location1,
+                                                   '/etc/pam.d/system-auth')
+
+    def restore_conf():
+        """ Restore """
+        execute_cmd(session_multihost, "cp -vf /etc/pam.d/system-auth_f_1949137 "
+                                       "/etc/pam.d/system-auth")
+        execute_cmd(session_multihost, "userdel -rf systest")
+
+    request.addfinalizer(restore_conf)
```

Comment:
You could reuse `create_localusers()` function with slight modifications to make it work and this would allow us to avoid maintaining another configuration function.

Reply:
create_localusers fixture are taking care of too many things, which we dont need this test case at all and its purpose is totally different from this test case

Reply:
Then try to be as generic as possible so that we can reuse this fixture in the future for other tests.

Reply:
Yes, one thing we can do is that delete this fixture and put these lines directly to that test case but if test case fails restoration or deletion of the users will not happen .

Reply:
Now fixture is generic and can be reused in other test case, and create_localusers fixture is reused in this test too.

Reply:
This looks better.